pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -654,32 +708,77 @@ func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Conte
-				ret.LastFailedRevisionErrors = []string{fmt.Sprintf("no detailed termination message, see `oc get -oyaml -n %q pods %q`", ns, name)}
+			if startupMonitor {
+				// this will be overridden again when the fallback pod is observed
+				ret.LastFailedRevisionErrors = append(ret.LastFailedRevisionErrors, "falling back to last-known-good revision")
+			} else {
+				ret.TargetRevision = 0 // stop installer retries, but keep running in startup-monitor case where we leave this to the startup-monitor to do
+				ret.LastFailedRevision = ret.TargetRevision
+
+				// increase failed count, but not in startup-monitor case where this is done when the fallback pod is observed below in staticPodStateFallback,
+				ret.LastFailedRevision = currNodeState.TargetRevision
+				ret.LastFailedReason = NodeStatusOperandFailedReason
+				if ret.LastFailedTime == nil || !ret.LastFailedTime.Time.Equal(ts) {
+					// avoid increasing the count continuously by comparing the fail time
+					failTime := metav1.NewTime(ts)
+					ret.LastFailedTime = &failTime
+					ret.LastFailedCount++
+				}
 			}
 
-			return ret, false, fmt.Sprintf("operand pod failed: %v", strings.Join(ret.LastFailedRevisionErrors, "\n")), nil
+			return ret, false, fmt.Sprintf("operand pod failed: %v", strings.Join(staticPodErrors, "\n")), nil
 
 		case staticPodStateReady:
-			rev := currNodeState.CurrentRevision
-			if currNodeState.TargetRevision > currNodeState.CurrentRevision {
-				rev = currNodeState.TargetRevision
+			if !startupMonitor {
+				rev := currNodeState.CurrentRevision
+				if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+					rev = currNodeState.TargetRevision
+				}
+				return &operatorv1.NodeStatus{
+					NodeName:        currNodeState.NodeName,
+					CurrentRevision: rev,
+				}, false, staticPodReason, nil
 			}
-			return &operatorv1.NodeStatus{
-				NodeName:        currNodeState.NodeName,
-				CurrentRevision: rev,
-			}, false, staticPodReason, nil
+
+			// in case of startup-monitor we leave marking a ready operand as current version to the startup-monitor.
+			//
+			// Two edge case:
+			// 1. startup-monitor is enabled. A new operand is rolled out with the startup-monitor. It will mark the new revision as current.
+			// 2. startup-monitor is disabled. A new operand is rolled out without the startup-monitor pod. The operator will now mark the revision as current.
+			// 3. startup-monitor is disabled, while a new operator WITH startup-monitor is rolled out. The upper revision override will mark the revision as current.
+
+			return currNodeState, false, staticPodReason, nil
+
+		case staticPodStateFallback:
+			ret := currNodeState.DeepCopy()
+			if ret.LastFailedTime == nil || !ret.LastFailedTime.Time.Equal(ts) { // do this only once per fallback
+				failTime := metav1.NewTime(ts)
+				ret.LastFailedTime = &failTime
+				ret.LastFailedRevision = ret.TargetRevision
+				ret.LastFailedRevisionErrors = staticPodErrors
+				ret.LastFailedCount++
```

Comment:
increasing only `LastFallbackCount` is not enough?

Reply:
> // lastFailedCount is how often the last failed revision failed. This includes failed installers and fallbacks to previous revisions.

I had a reason for this but forgot. We can clean that up in a follow-up (also in api repo) if we want to.

Reply:
I remember. because the lastFailedCount is used in the installer name. But we could just add lastFailedCount and lastFallbackCount.

Reply:
let me do that and we clean up the api docs later (it's in status, nobody cares).